x/attribute/keeper/msg_server.go

```diff
@@ -122,6 +126,40 @@ func (k msgServer) UpdateAttribute(goCtx context.Context, msg *types.MsgUpdateAt
 	return &types.MsgUpdateAttributeResponse{}, nil
 }
 
+func (k msgServer) UpdateAttributeExpiration(goCtx context.Context, msg *types.MsgUpdateAttributeExpirationRequest) (*types.MsgUpdateAttributeExpirationResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	attribute := types.Attribute{
+		Address: msg.Account,
+		Name:    msg.Name,
+		Value:   msg.Value,
+	}
+
+	ownerAddr, err := sdk.AccAddressFromBech32(msg.Owner)
+	if err != nil {
+		return nil, err
+	}
+
+	if msg.ExpirationDate != nil && msg.ExpirationDate.Unix() < ctx.BlockTime().Unix() {
+		return nil, fmt.Errorf("attribute expiration date %v is before block time of %v", msg.ExpirationDate.UTC(), ctx.BlockTime().UTC())
+	}
```

Comment:
I see this in several places. It's probably worthwhile to extract the logic to a `IsBeforeBlockTime(ctx, msg.ExpirationDate) bool` function and the error generation to a `ValidateExpirationBlockTime(ctx, msg.ExpirationDate) error` function.